### DISCUSSION TITLE
Help with XAMMP

### DISCUSSION THREAD

**Discussion by JOROLOJI** [mariuz/firebirdwebadmin#32] [Q&A]
Hello,

I have this error:
Fatal error: Uncaught Error: Undefined constant "IBASE_COMMITTED" in C:\xampp\htdocs\firebirdwebadmin\inc\configuration.inc.php:137 Stack trace: #0 C:\xampp\htdocs\firebirdwebadmin\inc\script_start.inc.php(8): require() #1 C:\xampp\htdocs\firebirdwebadmin\database.php(9): require('C:\\xampp\\htdocs...') #2 {main} thrown in C:\xampp\htdocs\firebirdwebadmin\inc\configuration.inc.php on line 137

Can you help me install in XAMPP with Windows 10 64bits?

**Comment by scozzaro** [comment]
use this config: <?php
// Purpose        basic config, set global constants
//                this is the only file that is included in every script
// Author         Lutz Brueckner <irie@gmx.de>
// Copyright      (c) 2000-2006 by Lutz Brueckner,
//                published under the terms of the GNU General Public Licence v.2,
//                see file LICENCE for details


//
// For the defines of paths you have to use slashes, even in a windows environment!
// i.e define('BINPATH', 'c:/firebird/bin/');
//

define('VERSION', '3.4.1');

// ============================================================
// === CONFIGURAZIONE PER FIREBIRD 5.0 SU WINDOWS/XAMPP ======
// ============================================================

// --- Percorso degli eseguibili Firebird (isql, gbak, ecc.) ---
// Firebird 5.0 su Windows (percorso standard)
define('BINPATH', 'C:/Program Files (x86)/Firebird\Firebird_5_0/');

// --- Directory temporanea (deve essere scrivibile da Apache) ---
define('TMPPATH', 'C:/xampp/tmp/');

// --- Impostazioni predefinite per il login al database ---
define('DEFAULT_USER',    'SYSDBA');
define('DEFAULT_DB',      'employee.fdb');
define('DEFAULT_PATH',    'C:/Program Files (x86)/Firebird\Firebird_5_0/examples/empbuild/');
define('DEFAULT_HOST',    'localhost');
define('DEFAULT_ROLE',    '');
define('DEFAULT_CACHE',   75);
define('DEFAULT_CHARSET', 'NONE');
define('DEFAULT_DIALECT', 3);
define('DEFAULT_SERVER',  'FB_5.0');               // 'FB_2.1', 'FB_2.5', 'FB_3.0', 'FB_4.0', 'FB_5.0'

define('PROTOCOL', 'http');                        // change to 'https' to use ssl

define('SP_LIMIT_NAME', '');

// --- Restrizioni di accesso ai database ---
// Se non vuoi restrizioni, lascia gli array vuoti
$ALLOWED_DIRS = array();
$ALLOWED_FILES = array();

// --- Directory per i backup (deve essere scrivibile) ---
define('BACKUP_DIR', 'C:/xampp/firebird_backup/');

// --- Lingua predefinita ---
define('LANGUAGE', 'english');

// --- Pannelli da nascondere (lascia vuoto per mostrarli tutti) ---
$HIDE_PANELS = array(
//                      'db_create',      // Create Database
//                      'db_delete',      // Delete Database
//                      'db_systable',    // System Tables
//                      'db_meta',        // Metadata
//                      'tb_show',        // View Tables
//                      'tb_create',      // Create New Table
//                      'tb_modify',      // Modify Table
//                      'tb_delete',      // Delete Table
//                      'acc_index',      // Indexes
//                      'acc_gen',        // Generators
//                      'acc_trigger',    // Triggers
//                      'acc_proc',       // Stored Procedures
//                      'acc_domain',     // Domains
//                      'acc_views',      // Views
//                      'acc_exc',        // Exceptions
//                      'acc_udf',        // User Defined Functions
//                      'sql_enter',      // Enter Command or Script
//                      'sql_output',     // Show Output
//                      'dt_enter',       // Enter Data
//                      'dt_export',      // Export Data
//                      'dt_import',      // Import Data
//                      'tb_watch',       // Watch Table
//                      'usr_user',       // Users
//                      'usr_role',       // Roles
//                      'usr_cust',       // Customizing
//                      'adm_server',     // Server Statistics
//                      'adm_dbstat',     // Database Statistics
//                      'adm_gfix',       // Database Maintenance
//                      'adm_backup',     // Backup
//                      'adm_restore'     // Restore
                     );

// --- Disabilita comandi SQL pericolosi ---
$SQL_DISABLE = array('CREATE DATABASE',   // disables creation of databases/schemas
                     'CREATE SCHEMA'
//                     'DROP'             // uncommenting this disables all DROP statements
//                     'DROP TABLE'       // uncommenting this disables the DROP TABLE statement
                     );

// --- Se TRUE, SYSDBA può fare tutto ignorando $HIDE_PANELS e $SQL_DISABLE ---
define('SYSDBA_GET_ALL', true);

// --- Chiedi conferma prima di cancellare ---
define('CONFIRM_DELETE', true);

// --- Dimensioni dell'area SQL ---
define('SQL_AREA_COLS', 80);
define('SQL_AREA_ROWS', 6);

// --- Altezza iframe in pixel ---
define('IFRAME_HEIGHT', 350);

// --- Salvataggio SQL nella sessione ---
define('SQL_MAXSAVE', 100);

// --- Numero di comandi nella cronologia SQL ---
define('SQL_HISTORY_SIZE', 25);

// --- Righe da mostrare nel pannello SQL output ---
define('SHOW_OUTPUT_ROWS', 100);

// --- Larghezza massima per i campi di input ---
define('DATA_MAXWIDTH', 50);

// --- Numero di righe per le FK Lookup ---
define('FKLOOKUP_ENTRIES', 1000);

// --- Lunghezza massima per una riga CSV ---
define('MAX_CSV_LINE', 50000);

// --- Righe predefinite nel pannello Watch ---
define('DEFAULT_ROWS', 25);

// --- Dimensioni finestra BLOB ---
define('BLOB_WINDOW_WIDTH', 600);
define('BLOB_WINDOW_HEIGHT', 800);

// --- Colori per la personalizzazione dell'interfaccia ---
define('COLOR_BACKGROUND',      '#FFFFFF');
define('COLOR_PANEL',           '#EEEEEE');
define('COLOR_AREA',            '#DDDDDD');
define('COLOR_HEADLINE',        '#CCCCCC');
define('COLOR_MENUBORDER',      '#BBBBBB');
define('COLOR_IFRAMEBORDER',    '#AAAAAA');
define('COLOR_IFRAMEBACKGROUND','#999999');
define('COLOR_LINK',            '#0000FF');
define('COLOR_LINKHOVER',       '#0000AA');
define('COLOR_SELECTEDROW',     '#FFFF00');
define('COLOR_SELECTEDINPUT',   '#00FF00');
define('COLOR_FIRSTROW',        '#F0F0F0');
define('COLOR_SECONDROW',       '#E0E0E0');

// --- Nome della sessione ---
define('SESSION_NAME', 'firebirdwebadmin');

// --- Parametri di transazione per fbird_trans() ---
define('TRANS_READ', IBASE_COMMITTED | IBASE_NOWAIT | IBASE_READ);
define('TRANS_WRITE', IBASE_COMMITTED | IBASE_NOWAIT | IBASE_WRITE);

// --- Redirezione lato server (FALSE) o client (TRUE) ---
define('META_REDIRECT', false);

// --- Opzioni di debug ---
define('DEBUG', false);
define('DEBUG_HTML', false);
define('DEBUG_COMMANDS', false);
define('DEBUG_FILES', true);

// --- Avvio della sessione ---
if ('' != SESSION_NAME && PHP_SAPI !== 'cli') {
    session_name(SESSION_NAME);
}

// --- Reporting errori in debug ---
if (DEBUG === true) {
    error_reporting(E_ALL | E_NOTICE | E_STRICT);
}
?>
